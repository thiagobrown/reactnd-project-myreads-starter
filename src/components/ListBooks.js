import React from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'

import BookShelf from './BookShelf'
import Loader from './Loader'
import { shelfs } from './../utils/constants'

const ListBooks = ({ books = [], loading = false, onUpdateBook }) => {

    return (
        <div className="list-books">
            <div className="list-books-title">
                <h1>MyReads</h1>
            </div>
            <div className="list-books-content">
                {
                    loading
                        ? <Loader />
                        : (
                            <div>
                                {shelfs.map(s => (
                                    <BookShelf key={s.shelf} title={s.title}
                                        books={books.filter(book => book.shelf === s.shelf)}
                                        onUpdateBook={onUpdateBook} />
                                ))}
                            </div>
                        )
                }
            </div>
            <div className="open-search">
                <Link to="/search" />
            </div>
        </div>
    )
}

ListBooks.propTypes = {
    books: PropTypes.array,
    loading: PropTypes.bool,
    onUpdateBook: PropTypes.func.isRequired
}

export default ListBooks;