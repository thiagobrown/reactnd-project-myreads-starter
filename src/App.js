import React, { Component } from 'react'
import { Route, Redirect, Switch } from 'react-router-dom'

import ListBooks from './components/ListBooks'
import SearchBooks from './components/SearchBooks'
import * as BooksAPI from './utils/BooksAPI'

import './App.css'

class App extends Component {
  state = {
    books: [],
    isLoading: false
  }

  componentDidMount() {
    this.setState({ isLoading: true });
    BooksAPI.getAll().then((books) => {
      this.setState({ books: books || [], isLoading: false });
    }).catch(error => {
      this.setState({ books: [], isLoading: false });
    });
  }

  onUpdateBook = (book, shelf) => {
    BooksAPI.update(book, shelf).then(resp => {
      book.shelf = shelf;
      this.setState(prevState => ({
        books: prevState.books.filter(b => b.id !== book.id).concat(book)
      }));
    });
  }

  render() {
    return (
      <div className="app">
        <Switch>
          <Route exact path="/" render={() => (
            <ListBooks
              books={this.state.books}
              loading={this.state.isLoading}
              onUpdateBook={this.onUpdateBook} />
          )} />
          <Route path="/search" render={() => (
            <SearchBooks
              books={this.state.books}
              onUpdateBook={this.onUpdateBook} />
          )} />
          <Redirect to="/" />
        </Switch>
      </div>
    )
  }
}

export default App;
