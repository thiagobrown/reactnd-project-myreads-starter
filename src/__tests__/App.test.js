import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import { MemoryRouter } from 'react-router-dom';

import App from './../App'
import ListBooks from './../components/ListBooks'

describe('Testing App Component', () => {
  let books = [{
    id: 1,
    title: 'Book One',
    authors: ['Authors One'],
    imageLinks: { thumbnail: 'url/book/one'},
    shelf: 'read'
  },{
    id: 2,
    title: 'Book Two',
    authors: ['Authors Two'],
    shelf: 'wantToRead'
  },{
    id: 3,
    title: 'Book Three',
    imageLinks: { thumbnail: 'url/book/three'},
    shelf: 'currentlyReading'
  }];

  const wrapper = shallow(<App />);
  
  it('render correctly', () => {
    wrapper.setState({ books: books, isLoading: false });
    expect(wrapper).toMatchSnapshot();
  });

  it('verify values of state/props in a component', () => {
    expect(wrapper.state().books).toBe(books);
    expect(wrapper.state().isLoading).toBe(false);
  });

  it('verify exists className app', () => {
    expect(wrapper.find('div').hasClass('app')).toEqual(true);
  });
});



