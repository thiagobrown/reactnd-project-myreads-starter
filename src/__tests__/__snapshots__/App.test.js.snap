exports[`Testing App Component render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
                exact={true}
                path="/"
                render={[Function]} />
        <Route
                path="/search"
                render={[Function]} />
        <Redirect
                push={false}
                to="/" />
</Switch>,
      "className": "app",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]} />,
          <Route
            path="/search"
            render={[Function]} />,
          <Redirect
            push={false}
            to="/" />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/search",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "push": false,
            "to": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]} />
          <Route
                    path="/search"
                    render={[Function]} />
          <Redirect
                    push={false}
                    to="/" />
</Switch>,
        "className": "app",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]} />,
            <Route
              path="/search"
              render={[Function]} />,
            <Redirect
              push={false}
              to="/" />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/search",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "push": false,
              "to": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
